### PR TITLE
Align CREATE SINK and CREATE SOURCE WITH options for Kerberos

### PR DESCRIPTION
**Pull request by msudberg**
Align CREATE SINK and CREATE SOURCE WITH options, add some missing options.

### CODE REVIEW

**Review by benesch — APPROVED** [review]

File: doc/user/layouts/partials/create-source/connector/kafka/with-options.html [hunk 1]
```diff
@@ -27,12 +28,14 @@
...
 Field | Value | Description
 ------|-------|------------
-`sasl_kerberos_keytab` | `text` | The absolute path to your keytab.
-`sasl_kerberos_kinit_cmd` | `text` | Shell command to refresh or acquire the client's Kerberos ticket.
-`sasl_kerberos_min_time_before_relogin` | `text` | Minimum time in milliseconds between key refresh attempts. Disable automatic key refresh by setting this property to 0.
-`sasl_kerberos_principal` | `text` | Materialize Kerberos principal name. Required for `sasl_plaintext`.
-`sasl_kerberos_service_name` | `text` | Kafka's service name on its host, i.e. the service principal name not including `/hostname@REALM`.
-`sasl_mechanisms` | `text` | The SASL mechanism to use for authentication. Currently, the only supported mechanisms are `'GSSAPI'` (the default) and `'PLAIN'`.
+`sasl_mechanisms` | `text` | The SASL mechanism to use for authentication. Currently, the only supported mechanisms are `GSSAPI` (the default) and `PLAIN`.
+`sasl_username` | `text` | Username required if `sasl_mechanisms` is `PLAIN`.
```
Comment (awang) [inline doc/user/layouts/partials/create-source/connector/kafka/with-options.html:32]: suggestion (feel free to reject/ignore): 

1. add parentheses:
> Username (required if sasl_mechanisms is PLAIN).
2. or drop `Username` altogether:
> (required if sasl_mechanisms is PLAIN)

I find that the "Username" and "Password is somewhat unnecessary, and the lack of punctuation between "Username" and "requirement" slightly hinders readability

**Review by awang — APPROVED** [review]
lgtm, thank you for doing this!

Observation:
we seem to combine "allowed values" into the "description" column, and sometimes it's not super obvious which we are describing.  Maybe (in the future) we can consider breaking "allowed values" out to it's own column (or combining it with `Value`), or adding some way of formatting/highlighting to distinguish it?

**Review by elindsey — COMMENTED** [review]
Looks good! A few minor thoughts 🙂

**Comment by msudberg** [comment]
> lgtm, thank you for doing this!
> 
> Observation:
> we seem to combine "allowed values" into the "description" column, and sometimes it's not super obvious which we are describing. Maybe (in the future) we can consider breaking "allowed values" out to it's own column (or combining it with `Value`), or adding some way of formatting/highlighting to distinguish it?

I think if we adjust the CSS to make the third page column a little narrower, we can do tables with more columns for that. I'd also like to standardize the way we represent default values. I'll create an issue forthis.

File: doc/user/content/sql/create-sink.md [hunk 2]
```diff
@@ -55,6 +55,9 @@ Field                | Value type | Description
 `partition_count`    | `int`      | Set the sink Kafka topic's partition count. This defaults to -1 (use the broker default).
 `replication_factor` | `int`      | Set the sink Kafka topic's replication factor. This defaults to -1 (use the broker default).
 `consistency`        | `boolean`  | Makes the sink emit additional [consistency metadata](#consistency-metadata). Only valid for Kafka sinks. This defaults to false.
+`security_protocol` | `text` | Use [`ssl`](#ssl-with-options) or, for [Kerberos](#kerberos-with-options), `sasl_plaintext`, `sasl-scram-sha-256`, or `sasl-sha-512` to connect to the Kafka cluster.
+`acks` | `text`| Specifies the number of replicas whose acknowledgement Kafka requires before returning to the client. Accepts values [-1,1000].
```
[the remark this answers is not in the document]
Comment (elindsey) [reply doc/user/content/sql/create-sink.md:59]: 1. -1/all is the default
2. Works for me!

File: doc/user/content/sql/create-sink.md [hunk 3]
```diff
@@ -55,6 +55,8 @@ Field                | Value type | Description
 `partition_count`    | `int`      | Set the sink Kafka topic's partition count. This defaults to -1 (use the broker default).
 `replication_factor` | `int`      | Set the sink Kafka topic's replication factor. This defaults to -1 (use the broker default).
 `consistency`        | `boolean`  | Makes the sink emit additional [consistency metadata](#consistency-metadata). Only valid for Kafka sinks. This defaults to false.
+`security_protocol` | `text` | Use [`ssl`](#ssl-with-options) or, for [Kerberos](#kerberos-with-options), `sasl_plaintext`, `sasl-scram-sha-256`, or `sasl-sha-512` to connect to the Kafka cluster.
+`acks` | `text`| Sets the number of kafka replicas that must acknowledge Materialize writes. Accepts values [-1,1000]. `-1` (the default) specifies all replicas.
```
Comment (elindsey) [inline doc/user/content/sql/create-sink.md:59]: kafka -> Kafka

**Review by elindsey — APPROVED** [review]

**Pull request merged**